lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -56,43 +71,36 @@ public TaskRuntime(IInjector taskInjector, string contextId, string taskId, Hear
 
                 // do not rethrow since this is benign
             }
+
             try
             {
-                heartBeatManager.EvaluatorSettings.NameClient = _injector.GetInstance<INameClient>();
```

Comment:
A `Task` alone should not modify a global setting. If the user really wants the setting to be global, they can submit the `INameClient` as a service. Submitting `INameClient` as a task still works, except we will be creating a new `NameClient` for each task, which is the correct thing to do in this case.